cmd/weaviate/main.go

```diff
@@ -0,0 +1,328 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"path/filepath"
+	"runtime/debug"
+	"strings"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/adapters/clients"
+	"github.com/weaviate/weaviate/adapters/repos/db"
+	"github.com/weaviate/weaviate/adapters/repos/db/inverted"
+	clusterweaviate "github.com/weaviate/weaviate/cluster"
+	"github.com/weaviate/weaviate/entities/replication"
+	"github.com/weaviate/weaviate/entities/vectorindex"
+	"github.com/weaviate/weaviate/exp/query"
+	"github.com/weaviate/weaviate/grpc/generated/protocol/v1"
+	"github.com/weaviate/weaviate/usecases/auth/authentication/apikey"
+	"github.com/weaviate/weaviate/usecases/auth/authentication/composer"
+	"github.com/weaviate/weaviate/usecases/auth/authentication/oidc"
+	"github.com/weaviate/weaviate/usecases/auth/authorization"
+	"github.com/weaviate/weaviate/usecases/cluster"
+	"github.com/weaviate/weaviate/usecases/config"
+	"github.com/weaviate/weaviate/usecases/memwatch"
+	"github.com/weaviate/weaviate/usecases/modules"
+	"github.com/weaviate/weaviate/usecases/objects"
+	"github.com/weaviate/weaviate/usecases/scaler"
+	"github.com/weaviate/weaviate/usecases/schema"
+	"github.com/weaviate/weaviate/usecases/traverser"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+
+	schemarepo "github.com/weaviate/weaviate/adapters/repos/schema"
+)
+
+const (
+	TargetQuerier = "querier"
+
+	// For mocking purpose
+	authAnonymousEnabled = true
+)
+
+// TODO: We want this to be part of original `cmd/weaviate-server`.
+// But for some reason that binary is auto-generated and I couldn't modify as I need. Hence separate binary for now
```

Comment:
For now i'd say it's fine to have it separated binary wise. We merge it later if we need but keeping it "experimental" for now the separation of binaries is OK